koku/providers/gcp/provider.py

```diff
@@ -24,24 +28,34 @@ def cost_usage_source_is_reachable(self, credentials, data_source):
         Verify that the GCP bucket exists and is reachable.
 
         Args:
-            credentials (object): not used; only present for interface compatibility
-            data_source (dict): dict containing name of GCP storage bucket
+            credentials (object): a dictionary containing project_id
+            data_source (dict): not used; only present for interface compatibility
 
         """
-        storage_client = storage.Client()
-        bucket = data_source.get("bucket")
         try:
-            bucket_info = storage_client.lookup_bucket(bucket)
-            if not bucket_info:
-                # if the lookup does not return anything, then this is an nonexistent bucket
-                key = "billing_source.bucket"
-                message = f"The provided GCP bucket {bucket} does not exist"
-                raise serializers.ValidationError(error_obj(key, message))
+            project = credentials.get("project_id")
+            gcp_credentials, _ = google.auth.default()
+            # https://github.com/googleapis/google-api-python-client/issues/299
+            service = discovery.build("cloudresourcemanager", "v1", credentials=gcp_credentials, cache_discovery=False)
+            check_permissions = {"permissions": REQUIRED_IAM_PERMISSIONS}
+            request = service.projects().testIamPermissions(resource=project, body=check_permissions)
+            response = request.execute()
+            permissions = response.get("permissions", [])
+
+            for required_permission in REQUIRED_IAM_PERMISSIONS:
+                if required_permission not in permissions:
+                    key = "authentication.project_id"
+                    err_msg = f"Improper IAM permissions: {permissions}."
+                    raise serializers.ValidationError(error_obj(key, err_msg))
```

Comment:
FYI: There is a list of user facing error strings that we can include for GCP when we complete the platform sources integration: https://github.com/project-koku/koku/blob/master/koku/providers/provider_errors.py.  We can surface those messages here.

Reply:
Ohhh. Thanks for pointing this out @dccurtis I didn't know this existed.